h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/matrix/SparseMatrixFactory.java

```diff
@@ -161,30 +154,37 @@ protected void map(int chunkIdx) {
             NestedArrayPointer rowHeaderPointer = new NestedArrayPointer(rwRow);
             NestedArrayPointer dataPointer = new NestedArrayPointer(nonZeroCount);
 
-            for(long i = _f.anyVec().espc()[chunk]; i < _f.anyVec().espc()[chunk+1]; i++) {
-                if (_w != null && _w.at(i) == 0) continue;
+            Chunk weightChunk = _w != null ? _w.chunkForChunkIdx(chunk) : null;
+            Chunk respChunk = _respVec.chunkForChunkIdx(chunk);
+            Chunk[] featChunks = new Chunk[_vecs.length];
```

Comment:
I think this array is not necessary we can do _vecs[i].chunkForChunkIdx(chunk) inside every for loop and save a bit fo memory, with same complexity

Reply:
I would (strongly) advise against that. The default `chunkForChunkIdx` is not super-trivial by itself (it even updates the last accessed time on on the associated Value in DKV). But the main reason is that this method can be and is overridden in WrappedVec implementations. The overriding implementations are typically much more expensive - they typically create a new instance of Chunk.